README.md

```diff
@@ -95,7 +95,7 @@ and the following Typescript types:
 - ```ts
   interface GoAsyncOptions<E extends Error = Error> {
     retries?: number; // Number of retries to attempt if the go callback is unsuccessful.
-    attemptTimeoutMs?: number; // The timeout for each attempt.
+    attemptTimeoutMs?: number | number[]; // The timeout for each attempt. Can provide an array for different timeouts for each attempt.
```

Comment:
This should also describe the behaviour when the length of this array doesn't match the number of `retries`. Either too many entries or too few

Reply:
Actually, wouldn't it be better to just ignore `retries` and use the length of the array as the source of truth? cc @Siegrift

Reply:
I haven't come up with a "perfect" configuration. I was thinking about allowing to pass array to `retries` (but that only configures retries and not the first call), separate property `attempts` accepting arrays with config for each attempt (too much work, doesn't work well with current way). 

I always prefer failing fast, but go should never throw. Failing the go with "configuration error" [is wierd](https://github.com/api3dao/promise-utils/issues/56#issuecomment-1209059799). I think it's OK to take the array length as source of truth.

I am now considering logging warnings for bad configuration (negative timeouts, this issue). Ofc, warnings would only appear in dev mode. WDYT?

Reply:
yeah, I don't think `go()` should throw so, I defaulted to the first element, it should be simple enough to default to the last element which is also intuitive

Reply:
>Logging a warning seems good to me. What is "dev mode" though? NODE_ENV !== 'production'? 

Yeah, but that feels like it should be done separately https://github.com/api3dao/promise-utils/issues/93

Reply:
> I defaulted to the first element, it should be simple enough to default to the last element which is also intuitive

Default to last seems a bit more intuitiv.

Reply:
Ok cool - so we're all in agreement then. Default to the last `attemptTimeoutMs` value if `retries > attemptTimeoutMs.length`. Please don't forget to document this as well

> feels like it should be done separately

agreed